functions/data.pp

```diff
@@ -0,0 +1,57 @@
+function gdm::data() {
```

Comment:
for only one conditional, a single file with an if statement seemed much simpler than 3 yaml files and a hiera.yaml